cachi2/core/package_managers/generic/main.py

```diff
@@ -69,7 +74,7 @@ def _resolve_generic_lockfile(lockfile_path: Path, output_dir: RootedPath) -> li
         Path.mkdir(Path(artifact.filename).parent, parents=True, exist_ok=True)
         to_download[str(artifact.download_url)] = artifact.filename
 
-    asyncio.run(async_download_files(to_download, get_config().concurrency_limit))
+    asyncio.run(async_download_files(files_to_download=to_download, concurrency_limit=get_config().concurrency_limit, ssl_context=ssl_context))
```

Comment:
nitpick: Passing the first 2 by keyword is an unrelated change, those are declared as positional, not optional, so it should either be changed in the called function itself (in a different commit though) or just drop the changes up to `ssl_context` which you need.